lib/redis-manager.js

```diff
@@ -0,0 +1,234 @@
+const redis = require('redis');
+const CacheManagerError = require('./cache-manager-error');
+
+/**
+*    RedisManager class
+*/
+class RedisManager {
+
+	get MS() {
+		return process.env.MICROSERVICE || 'node';
+	}
+
+	/**
+     * Get the Redis config
+     * @return {object}
+     */
+	get config() {
+		if(!this._config)
+			this.cacheConfig();
+
+		return this._config;
+	}
+
+	set keyPrefix(prefix) {
+		this._keyPrefix = prefix;
+	}
+
+	get keyPrefix() {
+		return this._keyPrefix;
+	}
+
+	getKey(key) {
+		return `${this.keyPrefix}${key}`;
+	}
+
+	/**
+     * Get the Redis Config JSON path
+	 * @returns {String} route path
+     */
+	get configPath() {
+		return path.join(process.cwd(), 'config/redis.json');
+	}
+
+	/**
+     * Cache the Redis config.
+	* @param {String} route Config JSON path
+     */
+	cacheConfig() {
+
+		let config;
+
+		try {
+			config = require(this.configPath);
+		} catch(error) {
+			throw new CacheManagerError('Invalid config path', CacheManagerError.codes.CONFIG_NOT_FOUND);
+		}
+
+		this._config = config;
+	}
+
+	/**
+     * Prepares the params, adding MS prefix     *
+    * @param {object} params The parameters
+    * @return {String} encoded parameters
+    */
+	_prepareParams(params) {
+		return md5(JSON.stringify({
+			_MS: this.MS,
+			...params
+		}));
+	}
+
+	/**
+     * Initialize a Redis prefix in order to be ready to use.
+     * @param {String} prefix Name of the client-prefix.
+     */
+	constructor(clientPrefix) {
+
+		this.keyPrefix = this.validClientPrefix(clientPrefix);
+		this.clients = [];
+		this.inited = false;
+
+		this.client = this.createClient();
+		this.promisify(this.client);
+	}
+
+	/**
+	 * Check the client-prefix
+	 * @param {String} prefix name of prefix.
+	 * @returns {String} prefix name.
+	 */
+	validClientPrefix(clientPrefix) {
+		if(typeof clientPrefix !== 'string')
+			throw new CacheManagerError('Invalid client-prefix.', CacheManagerError.codes.INVALID_PREFIX);
+		return clientPrefix;
+	}
+
+	/**
+     * Create a redis prefix
+     * @param {object} options - Redis prefix options
+     * @return {object} redis client
+     */
+	createClient(options = {}) {
+		const { host, port } = this.configServer();
+
+		const defaults = {
+			host,
+			port
+		};
+
+		const client = this.clientRedis(defaults, options);
+
+		client.on('connect', () => {
+			logger.info(`Redis - Client-Prefix: ${this.keyPrefix} | connected to ${host}:${port}`);
+			this.inited = true;
+		});
+
+		client.on('error', err => logger.error(err.message));
+
+		client.on('reconnecting', () => {
+			logger.warn(`Redis - Client-Prefix: ${this.keyPrefix} | reconnecting`);
+		});
+
+		this.clients.push(client); // for close latter
+
+		return client;
+	}
+
+	/**
+	 * set the host and port values for the connect redis server
+	 * @returns {Object} port and host values for the connection
+	 */
+	configServer() {
+		const host = this.config.host || 'localhost';
+		const port = this.config.port || 6739;
```

Comment:
```suggestion
		const port = this.config.port || 6379;
```
El port default de redis es `6379`